Core/src/IfcGeometryConverter/GeomUtils.h

```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
 				const char* what() const throw() { return s.c_str(); }
 			};
 
+			/*!	\brief Helper functions for calculations of geometries
+
+			*/
 			class GeomUtils {
 			public:
 
+				/**********************************************************************************************/
+				/*! \brief Compare two doubles for equality to the custom precision
+				*
+				* \param[in] first					The first value
+				* \param[in] second					The second value
+				* \param[in] precision				Precision of model
+				*
+				* \return							True, if the absolute difference is smaller than the precision. False otherwise
+				*/
+				static bool GeomUtils::areEqual(const double first, 
+					const double second, const double precision);
+
+				/**********************************************************************************************/
+				/*! \brief Calculates a geometric center of the plane figure. 
+				*
+				* \param[in] polygon				Polygon, which contains points of a plane figure. 
+				*
+				* \return							Geometric center of the plane figure. 
+				*/
 				static carve::geom::vector<3> computePolygonCentroid(
-					const std::vector<carve::geom::vector<3> >& polygon);
+					const std::vector<carve::geom::vector<3>>& polygon);
+
+				/**********************************************************************************************/
+				/*! \brief Calculates a normal of the plane using points of the polygon on this plane.
+				*
+				* \param[in] polygon				Polygon with points (in 3D) on the plane. 
+				*
+				* \return							Normal vector of the plane. 
+				*/
 				static carve::geom::vector<3> computePolygonNormal(
-					const std::vector<carve::geom::vector<3> >& polygon);
+					const std::vector<carve::geom::vector<3>>& polygon);
+
+				/**********************************************************************************************/
+				/*! \brief Calculates a normal of the plane using points of the polygon on this plane.
+				*
+				* \param[in] polygon				Polygon with points (in 2D) on the plane.
+				*
+				* \return							Normal vector of the plane.
+				*/
 				static carve::geom::vector<3> computePolygon2DNormal(
-					const std::vector<carve::geom::vector<2> >& polygon);
-
+					const std::vector<carve::geom::vector<2>>& polygon);
+
+				/**********************************************************************************************/
+				/*! \brief Calculates coordinates of the intersection point.
+				*
+				* \param[in] initialPointSeg1		Initial point of the first intersecting segment.
+				* \param[in] terminalPointSeg1		Terminal point of the first intersecting segment.
+				* \param[in] initialPointSeg2		Initial point of the second intersecting segment, which describes the direction of the line.
+				* \param[in] terminalPointSeg2		Terminal point of the second intersecting segment, which describes the direction of the line.

```

Comment:
```suggestion
				* \param[in] terminalPointSeg2		2nd line segment, 2nd point.
```

Apply similarly for all comments of this sort: use (initial, terminal) for lines that are of fixed length, and (1st, 2nd) for lines that extend into infinity. 

I'd also include a comment in the description of the function, that the first/second line are to be taken as limited in length by the points or infinite.

Reply:
Resolved: e82ca1a